From f6c1eaeb597207bcdb0d89179a7bd246efd444e6 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Sun, 21 May 2017 19:32:57 +0200
Subject: [PATCH] SMBFile: Use 1MB chunks

---
 xbmc/filesystem/SMBFile.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/filesystem/SMBFile.h b/xbmc/filesystem/SMBFile.h
index 5fb5323..ee192ce 100644
--- a/xbmc/filesystem/SMBFile.h
+++ b/xbmc/filesystem/SMBFile.h
@@ -89,7 +89,7 @@ public:
   virtual bool OpenForWrite(const CURL& url, bool bOverWrite = false);
   virtual bool Delete(const CURL& url);
   virtual bool Rename(const CURL& url, const CURL& urlnew);
-  virtual int GetChunkSize() { return 1; }
+  virtual int GetChunkSize() { return 1024 * 1024; }
   virtual int IoControl(EIoControl request, void* param);
 
 protected:
-- 
2.0.1

