From 4d66410f49b523639cf88b1892adfa3abc0d60fc Mon Sep 17 00:00:00 2001
From: fritsch <fritsch@kodi.tv>
Date: Sat, 21 Jan 2017 17:59:43 +0100
Subject: [PATCH 6/9] Increase CHUNK_SIZE to 128 KB

---
 xbmc/filesystem/FileCache.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/filesystem/FileCache.cpp b/xbmc/filesystem/FileCache.cpp
index 4519266..331960e 100644
--- a/xbmc/filesystem/FileCache.cpp
+++ b/xbmc/filesystem/FileCache.cpp
@@ -43,7 +43,7 @@
 
 using namespace XFILE;
 
-#define READ_CACHE_CHUNK_SIZE (64*1024)
+#define READ_CACHE_CHUNK_SIZE (128*1024)
 
 class CWriteRate
 {
-- 
2.0.1

