From 3fad56901de8651c821aa0c94c0c3215401a181b Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Thu, 20 Apr 2017 18:01:16 +0100
Subject: [PATCH 8/9] DVDDemuxFFMPEG: Increase FFMPEG_FILE_BUFFER_SIZE to 128K

---
 xbmc/cores/FFmpeg.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/FFmpeg.h b/xbmc/cores/FFmpeg.h
index b7d25bf..2643d1f 100644
--- a/xbmc/cores/FFmpeg.h
+++ b/xbmc/cores/FFmpeg.h
@@ -65,4 +65,4 @@ public:
   int level;
 };
 
-#define FFMPEG_FILE_BUFFER_SIZE   32768 // default reading size for ffmpeg
+#define FFMPEG_FILE_BUFFER_SIZE   131072 // default reading size for ffmpeg
-- 
2.0.1

