From ec7fad1340916e916c5e387a09908d73148ab226 Mon Sep 17 00:00:00 2001
From: fritsch <fritsch@kodi.tv>
Date: Sat, 21 Jan 2017 17:59:43 +0100
Subject: [PATCH] Increase CHUNK_SIZE to 128 KB

---
 xbmc/filesystem/FileCache.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/filesystem/FileCache.cpp b/xbmc/filesystem/FileCache.cpp
index afea2fe..26598f7 100644
--- a/xbmc/filesystem/FileCache.cpp
+++ b/xbmc/filesystem/FileCache.cpp
@@ -43,7 +43,7 @@
 
 using namespace XFILE;
 
-#define READ_CACHE_CHUNK_SIZE (64*1024)
+#define READ_CACHE_CHUNK_SIZE (128*1024)
 
 class CWriteRate
 {
