From 92c29358c4d820aa473c38880ee1fe8640e19feb Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 8 May 2018 19:53:40 +0200
Subject: [PATCH] gbm: select valid connector

---
 xbmc/windowing/gbm/DRMUtils.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/xbmc/windowing/gbm/DRMUtils.cpp b/xbmc/windowing/gbm/DRMUtils.cpp
index 912a789..c692cdf 100644
--- a/xbmc/windowing/gbm/DRMUtils.cpp
+++ b/xbmc/windowing/gbm/DRMUtils.cpp
@@ -145,7 +145,10 @@ bool CDRMUtils::GetConnector()
   {
     m_connector->connector = drmModeGetConnector(m_fd,
                                                       m_drm_resources->connectors[i]);
-    if(m_connector->connector->connection == DRM_MODE_CONNECTED)
+
+    /* Connectors must be connected, but also configured with valid encoders and CRTCs */
+    if (m_connector->connector->encoder_id &&
+        m_connector->connector->connection == DRM_MODE_CONNECTED)
     {
       CLog::Log(LOGDEBUG, "CDRMUtils::%s - found connector: %d", __FUNCTION__,
                                                                  m_connector->connector->connector_id);
-- 
2.0.1

