From f2e200edd44f83511aa966f0dcde789153cc647a Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Wed, 23 May 2018 20:04:40 -0700
Subject: [PATCH 5/6] libavcodec: v4l2m2m: use get_format callback from
 application

---
 libavcodec/v4l2_m2m_dec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
index 1bfd11e..d524fd2 100644
--- a/libavcodec/v4l2_m2m_dec.c
+++ b/libavcodec/v4l2_m2m_dec.c
@@ -200,7 +200,7 @@ static av_cold int v4l2_decode_init(AVCodecContext *avctx)
      *   - the DRM frame format is passed in the DRM frame descriptor layer.
      *       check the v4l2_get_drm_frame function.
      */
-    if (avctx->pix_fmt == AV_PIX_FMT_DRM_PRIME)
+    if (ff_get_format(avctx, avctx->codec->pix_fmts) == AV_PIX_FMT_DRM_PRIME)
         s->output_drm = 1;
 
     ret = ff_v4l2_m2m_codec_init(avctx);
-- 
2.0.1

