From 63d195be61499db0ba278cafdce5eefe0f0c2971 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Sat, 2 Jun 2018 09:36:52 -0700
Subject: [PATCH 6/6] libavcodec: v4l2m2m: allow using multiplane formats

---
 libavcodec/v4l2_buffers.c | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/libavcodec/v4l2_buffers.c b/libavcodec/v4l2_buffers.c
index a14c5ab..c12cf1f 100644
--- a/libavcodec/v4l2_buffers.c
+++ b/libavcodec/v4l2_buffers.c
@@ -215,21 +215,24 @@ static uint8_t * v4l2_get_drm_frame(V4L2Buffer *avbuf)
     drm_desc->nb_layers = 1;
 
     layer = &drm_desc->layers[0];
-    layer->planes[0].object_index = 0;
-    layer->planes[0].offset = 0;
-    layer->planes[0].pitch = avbuf->plane_info[0].bytesperline;
+
+    for (int i = 0; i < avbuf->num_planes; i++) {
+        layer->planes[i].object_index = i;
+        layer->planes[i].offset = 0;
+        layer->planes[i].pitch = avbuf->plane_info[i].bytesperline;
+    }
 
     switch (avbuf->context->av_pix_fmt) {
     case AV_PIX_FMT_NV12:
     case AV_PIX_FMT_NV21:
 
-        if (avbuf->num_planes > 1)
-            break;
-
         layer->format = avbuf->context->av_pix_fmt == AV_PIX_FMT_NV12 ?
             DRM_FORMAT_NV12 : DRM_FORMAT_NV21;
         layer->nb_planes = 2;
 
+        if (avbuf->num_planes > 1)
+            break;
+
         layer->planes[1].object_index = 0;
         layer->planes[1].offset = avbuf->plane_info[0].bytesperline *
             avbuf->context->format.fmt.pix_mp.height;
@@ -238,12 +241,12 @@ static uint8_t * v4l2_get_drm_frame(V4L2Buffer *avbuf)
 
     case AV_PIX_FMT_YUV420P:
 
-        if (avbuf->num_planes > 1)
-            break;
-
         layer->format = DRM_FORMAT_YUV420;
         layer->nb_planes = 3;
 
+        if (avbuf->num_planes > 1)
+            break;
+
         layer->planes[1].object_index = 0;
         layer->planes[1].offset = avbuf->plane_info[0].bytesperline *
             avbuf->context->format.fmt.pix_mp.height;
-- 
2.0.1

